### PR TITLE
docs(rumqttd): Fix typos in segments and server

### PR DESCRIPTION
**Pull request by sgasse**
Fix some typos in docstrings and comments in rumqttd in the modules `segements` and `server`.

## Type of change

 - Miscellaneous (related to maintanance)

## Checklist:

- [x] No code changed so no formatting needed.
- [x] Only fixing typos thus no entry in CHANGELOG.

### CODE REVIEW

File: rumqttd/src/segments/mod.rs [hunk 1]
```diff
@@ -24,36 +24,36 @@ pub trait Storage {
...
 /// - limit on number of segments stored in disk
 ///
-/// When the active_segment if filled up, we move it to memory segments and empty it for new logs.
-/// When limit on number of memory segments is reached, we move a single segment from memory onto
-/// the disk. When limit on number of elements on disk is also reached, we move a single segment
-/// from memory onto the disk.
+/// When the active_segment is filled up, we move it to memory segments and empty it for new logs.
+/// When the limit on number of memory segments is reached, we move a single segment from memory
+/// onto the disk. When the limit on the number of elements on disk is also reached, we move a
+/// single segment from memory onto the disk.
```
Comment (henil) [inline rumqttd/src/segments/mod.rs:30]:
```suggestion
/// When the active_segment is filled up, we move it to memory segments and empty it for new logs.
/// When the limit on number of memory segments is reached, We remove the oldest segment from 
/// memory segments.
```

File: rumqttd/src/segments/mod.rs [hunk 2]
```diff
@@ -24,36 +24,36 @@ pub trait Storage {
...
-///      exceed file's existing size
+///    - `index` & `segment` - everything directly related to files, no bounds check except when
+///      bounds exceed file's existing size.
 ///    - `chunk` - abstraction to deal with index and segment combined. Basically we only need
-///      stuff from segment file, and thus hides away the index file under abstraction.
+///      stuff from segment file, and thus we hide away the index file under this abstraction.
 ///    - `segment` - abstracts away the disk and memory segments for ease of access.
-///    - 'disk_handle' - hold onto the things needed when dealing with disk, for example the
+///    - 'disk_handle' - holds on to the things needed when dealing with disk, for example the
 ///      hashing struct, path, files with invalid names etc.
```
Comment (henil) [inline rumqttd/src/segments/mod.rs:51]: remove `disk_handle` part

File: rumqttd/src/segments/mod.rs [hunk 3]
```diff
@@ -67,7 +67,7 @@ impl<T> CommitLog<T>
 where
     T: Storage + Clone,
 {
-    /// Create a new `CommitLog` with the given contraints. If `None` if passed in for `disk`
+    /// Create a new `CommitLog` with the given contraints. If `None` is passed in for `disk`
     /// parameter, no disk persistence is provided. If `max_mem_segments` is 0, then only the
     /// active segment is maintained.
     ///
```
Comment (henil) [inline rumqttd/src/segments/mod.rs:73]: remove `disk` part from here.

**Review by henil — COMMENTED** [review]
we had disk segments previously but they where removed in past, now we only have memory segments.

so can you remove mentions related to `disk_segments` or `disks` from `segments` module as well

**Comment by sgasse** [comment]
Ah thanks @henil , I somehow overlooked that the disk part is no longer implemented. Now I removed all mentioning of `disk` in the file and rephrased the docstrings accordingly.

**Review by henil — COMMENTED** [review]
LGTM

**Comment by henil** [comment]
Thank you @sgasse :tada: !

**Pull request merged**